cvat/apps/engine/views.py

```diff
@@ -531,17 +531,18 @@ def get_upload_dir(self):
             return self._object.get_tmp_dirname()
         elif 'backup' in self.action:
             return backup.get_backup_dirname()
-        return ""
+        assert False
 
     def upload_finished(self, request):
         if self.action == 'dataset':
             format_name = request.query_params.get("format", "")
             filename = request.query_params.get("filename", "")
             conv_mask_to_poly = to_bool(request.query_params.get('conv_mask_to_poly', True))
             tmp_dir = self._object.get_tmp_dirname()
-            uploaded_file = None
-            if os.path.isfile(os.path.join(tmp_dir, filename)):
-                uploaded_file = os.path.join(tmp_dir, filename)
+            uploaded_file = os.path.join(tmp_dir, filename)
+            if not os.path.isfile(uploaded_file):
+                uploaded_file = None
```

Comment:
- It looks like it should be a 400 error instead. 
- There should be some path validation, like one in `is_valid_uploaded_file_name` of `mixins.py`. For backups as well.

Reply:
I think this comment should be addressed in a separate PR since it's not the purpose of this PR. I just changed a bit of code to not call `os.path.join` twice, the logic has not been changed.